src/Makefile.test.include

```diff
@@ -28,6 +27,8 @@ FUZZ_TARGETS = \
 
 if ENABLE_FUZZ
 noinst_PROGRAMS += $(FUZZ_TARGETS:=)
+else
+bin_PROGRAMS += test/test_bitcoin
```

Comment:
In commit "qa: Add test/fuzz/test_runner.py" (fa7ca8ef58bf3e3b91d1f5a67fa42008e63b1f7b)

@MarcoFalke  Is this no longer building test_bitcoin if ENABLE_FUZZ is true? I wouldn't expect ENABLE_FUZZ to affect this, but if this is correct it'd be helpful to have a comment here explaining what this is doing.

Reply:
Yes, I think there is no reason to build `test_bitcoin` if you want to build the fuzz tests. Am I missing something?

Regardless of that, it would result in linker errors later in the build process, since you can't link libfuzzer to something that has a main function.

Reply:
Oh ok, I don't know really anything about fuzzing, and I assumed enabling a fuzzing option would just build some new binaries, not affect existing binaries in any way. But I guess due to autotools lack of support for side-by-side build configurations, ENABLE_FUZZ affects every existing binary we build, as well as building new binaries? Probably not worth adding a comment here just for `test_bitcoin` in this case.

Reply:
Yes, ideally it should disable all other binaries.

Currently we hack around that by specifying it manually:

https://github.com/bitcoin/bitcoin/blob/169dced9a42bd741b3265adee23e6a8d1f852227/.travis.yml#L146

Reply:
It is surprising that --enable-fuzz silently stops building the test binary, especially as config.log continues to claim that test_bitcoin is being built.

Reply:
> It is surprising that --enable-fuzz silently stops building the test binary

Might be fixed by #19388, which is blocked on #20560 . Though when building and linking with a fuzz engine, the test_bitcoin binary won't have any meaning. Not sure if it will even compile/link.

> config.log continues to claim that test_bitcoin is being built.

I don't really understand build systems, so if this is an issue, I hope that someone will fix it

Reply:
I'll open a PR to change config to output something like
```
configure:27478: checking whether to build test_bitcoin-qt
configure:27481: result: no, because fuzzing is enabled
configure:27499: checking whether to build test_bitcoin
configure:27502: result: no, because fuzzing is enabled
```
It's not really surprising that enabling fuzzing would break all the other binaries, I was just misled by the config output.

Reply:
https://github.com/bitcoin/bitcoin/pull/20609